econml/dowhy.py

```diff
@@ -0,0 +1,222 @@
+            missing_nodes_as_confounders=missing_nodes_as_confounders
+        )
+        self.identified_estimand_ = self.dowhy_.identify_effect(proceed_when_unidentifiable=True)
+        method_name = "backdoor." + self._cate_estimator.__module__ + "." + self._cate_estimator.__class__.__name__
+        init_params = {}
+        for p in self._get_params():
+            init_params[p] = getattr(self._cate_estimator, p)
+        self.estimate_ = self.dowhy_.estimate_effect(self.identified_estimand_,
+                                                     method_name=method_name,
+                                                     control_value=control_value,
+                                                     treatment_value=treatment_value,
+                                                     target_units=target_units,
+                                                     method_params={
+                                                         "init_params": init_params,
+                                                         "fit_params": kwargs,
+                                                     },
+                                                     )
+        return self
+
+    def refute_estimate(self, *, method_name, **kwargs):
+        """Refute an estimated causal effect.
+
+        If method_name is provided, uses the provided method. In the future, we may support automatic
+        selection of suitable refutation tests.
+        Following refutation methods are supported:
+            * Adding a randomly-generated confounder: "random_common_cause"
+            * Adding a confounder that is associated with both treatment and outcome: "add_unobserved_common_cause"
+            * Replacing the treatment with a placebo (random) variable): "placebo_treatment_refuter"
+            * Removing a random subset of the data: "data_subset_refuter"
+
+        Parameters
+        ----------
+        method_name: string
+            Name of the refutation method
+        kwargs: optional
+            Additional arguments that are passed directly to the refutation method.
+            Can specify a random seed here to ensure reproducible results ('random_seed' parameter).
+            For method-specific parameters, consult the documentation for the specific method.
+            All refutation methods are in the causal_refuters subpackage.
+
+        Returns
+        -------
+        RefuteResult: an instance of the RefuteResult class
+        """
+        return self.dowhy_.refute_estimate(
+            self.identified_estimand_, self.estimate_, method_name=method_name, **kwargs
+        )
+
+    # We don't allow user to call refit_final from this class, since internally dowhy effect estimate will only update
+    # cate estimator but not the effect.
+    def refit_final(self, inference=None):
+        raise AttributeError(
+            "Method refit_final is not allowed here! Please call it from Cate Estimator directly! ")
```

Comment:
I'm not sure that we want to recommend calling it on the estimator directly, since (unless I'm misunderstanding) this wrapper will still not function as expected after refitting.

Reply:
Yes, the logic the user should follow is 
```
est=LinearDML()
est_dowhy=est.dowhy.fit()
```
I mean if you want to call refit, please call
```
est.fit()
est.refit_final()
```

Any suggestion on writing a better error message maybe?

Reply:
Well, the question to me is whether there's really a refit story that makes sense, or whether we should just say it is unsupported and not suggest any workaround other than doing a full fit instead.

I guess my question is, let's say I do

```python3
est = LinearDML()
dw = est.dowhy_
dw.fit(...)
est.refit_final()
```

It seems like the current error message suggests this is the right way to do things, but it's still not okay to try to call `dw.refute`, etc., right, because its internal estimator is out of date? Is there some way to sync the refitted estimator back up to the dowhy wrapper?

If not, then I think the error message should just be

```suggestion
            "Method refit_final is not allowed through a dowhy object; please perform a full fit instead.")
```